team-01/src/main/java/ubc/cosc322/MonteCarloPlayer.java

```diff
@@ -74,19 +74,45 @@ private TreeNode getBestNode(TreeNode root) {
 		return null;
 	}
 
-	private void backpropagate(TreeNode current, boolean won) {
+	private void backpropagate(TreeNode current, int winner) {
 		while (current != null) {
-			if (won) {
+			if (current.state.localPlayer == winner) {
 				current.wins++;
 			}
 			current.visits++;
 			current = current.parent;
 		}
 	}
 
-	// TODO
-	private boolean playthrough(TreeNode current) {
-		return true;
+	private int playthrough(TreeNode current) {
+		AmazonsLocalBoard boardCopy = current.state.copy();
+		TreeNode simRoot = new TreeNode(root.state, root.getAction(), root.parent);
+		int winner = Integer.MIN_VALUE;
+
+		while (winner < 0) {
+			ArrayList<AmazonsAction> actions = actionFactory.getActions(boardCopy);
+			//Check win conditions
```

Comment:
One newline before comments like this really helps readability. Also, don't forget to lint.